numpy/core/tests/test_dtype.py

```diff
@@ -88,6 +88,19 @@ def test_invalid_types(self):
             assert_raises(TypeError, np.dtype, 'q8')
             assert_raises(TypeError, np.dtype, 'Q8')
 
+    @pytest.mark.parametrize(
+        ['operator', 'expected'],
+        [('==', False), ('!=', True),
+        ('>', False), ('<', False),
+        ('>=', False), ('<=', False)])
+    def test_richcompare_invalid_dtype(self, operator, expected):
+        # Make sure objects that cannot be converted to valid
+        # dtypes results in False when compared to valid dtypes.
+        # Here 7 cannot be converted to dtype. No exceptions should be raised
+
+        assert eval(f"np.dtype(np.int32) {operator} 7") == expected,\
+                f"dtype richcompare failed for {operator}"
```

Comment:
I suppose you used eval, because you were not aware of `import operator; operator.eq`, etc.?

Reply:
Oh thanks! I did not know of this. Shall replace it.